pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -2126,6 +2126,18 @@ double Object_getDoubleValue(JSOBJ Py_UNUSED(obj), JSONTypeContext *tc) {
     return GET_TC(tc)->doubleValue;
 }
 
+const char *Object_getBigNumStringValue(JSOBJ obj, JSONTypeContext *tc, 
+                                    size_t *_outLen) {
+    PyObject* repr = PyObject_Repr(obj);
+    PyObject* str = PyUnicode_AsEncodedString(repr, "utf-8", "~E~");
+    char *bytes = PyBytes_AS_STRING(str);
```

Comment:
You will want PyUnicode_AsUTF8AndSize here. You can pass outLen as the second argument